astropy/nddata/utils.py

```diff
@@ -25,6 +28,16 @@
     "Cutout2D",
 ]
 
+if TYPE_CHECKING:
+    from numbers import Number
```

Comment:
What's the reason for using [`Number`](https://docs.python.org/3/library/numbers.html#numbers.Number) here? There's a few places in `units` where [`Fraction`](https://docs.python.org/3/library/fractions.html#fractions.Fraction) is a valid input type, so we use [`Real`](https://docs.python.org/3/library/numbers.html#numbers.Real) as a type annotation because of that, but is [`float`](https://docs.python.org/3/library/functions.html#float) not good enough in this file?

Reply:
I used `Number` in places where I assumed complex values could be valid, but if I'm wrong then `float` would be perfectly fine instead.

Reply:
Even though `float` is not a subclass of `complex`, [the `complex` type does encompass `float`](https://peps.python.org/pep-0484/#the-numeric-tower). If the annotations should explicitly allow `numpy` datatypes then consider using [`numbers.Complex`](https://docs.python.org/3/library/numbers.html#numbers.Complex) instead.